### PR TITLE
AB#40576 Migrate remaining emails to Notify

### PR DESCRIPTION
**Pull request by aidayounesi**
Would you mind also reviewing the new Notify templates in "GeoNotification" directory in our Notify dashboard

### CODE REVIEW

File: GenderPayGap.WebUI/Classes/Extensions/Extensions.cs [hunk 1]
```diff
@@ -130,10 +131,10 @@ public static User FindUser(this IDataRepository repository, IPrincipal principa
 
         #region Registraion Helpers
 
-        public static async Task<bool> SendVerifyEmailAsync(this Controller controller, string emailAddress, string verifyCode)
+        public static void SendVerifyEmail(this Controller controller, string emailAddress, string verifyCode)
         {
             string verifyUrl = controller.Url.Action("VerifyEmail", "Register", new {code = verifyCode}, "https");
-            return await Emails.SendCreateAccountPendingVerificationAsync(verifyUrl, emailAddress);
+            EmailSendingService.SendCreateAccountPendingVerificationEmail(emailAddress, verifyUrl);
```
Comment (jamesgriff) [inline GenderPayGap.WebUI/Classes/Extensions/Extensions.cs:137]: If this method is only used by the controller, then you might as well inline the method - it's only 2 lines and it's a bit confusing how it's currently an extension method

File: GenderPayGap.WebUI/Controllers/RegisterController.SignUp.cs [hunk 2]
```diff
@@ -53,10 +53,7 @@ private async Task<string> SendVerifyCodeAsync(User currentUser)
             try
             {
                 string verifyCode = Encryption.EncryptQuerystring(currentUser.UserId + ":" + currentUser.Created.ToSmallDateTime());
-                if (!await this.SendVerifyEmailAsync(currentUser.EmailAddress, verifyCode))
-                {
-                    return null;
-                }
+                this.SendVerifyEmail(currentUser.EmailAddress, verifyCode);
```
Comment (jamesgriff) [inline GenderPayGap.WebUI/Controllers/RegisterController.SignUp.cs:56]: See comment in `Extensions.cs`.

This line:
```csharp
this.SendVerifyEmail(currentUser.EmailAddress, verifyCode);
```

could become:
```csharp
string verifyUrl = Url.Action("VerifyEmail", "Register", new {code = verifyCode}, "https");
EmailSendingService.SendCreateAccountPendingVerificationEmail(emailAddress, verifyUrl);
```

File: GenderPayGap.WebUI/Services/EmailSendingService.cs [hunk 3]
```diff
@@ -116,6 +116,21 @@ public static async void SendScopeChangeOutEmail(string emailAddress, string org
 
             await AddEmailToQueue(notifyEmail);
         }
+        public static async void SendCreateAccountPendingVerificationEmail(string emailAddress, string url)
+        {
+            var personalisation = new Dictionary<string, dynamic> {
+                {"url", url}, {"Environment", Config.IsProduction() ? "" : $"[{Config.EnvironmentName}] "}
+            };
```
Comment (jamesgriff) [inline GenderPayGap.WebUI/Services/EmailSendingService.cs:123]: I'd prefer these two dictionary entries on separate lines (although I know the auto-formatted sometimes puts them on the same line!) - for a dictionary, I think it makes it clearer that we're adding two entries

i.e.
```csharp
var personalisation = new Dictionary<string, dynamic> {
    {"url", url},
    {"Environment", Config.IsProduction() ? "" : $"[{Config.EnvironmentName}] "}
};
```

File: GenderPayGap.Core/Enums.cs [hunk 4]
```diff
@@ -293,7 +293,6 @@ public static class Filenames
         public const string BadSicLog = "BadSicLog.csv";
         public const string ManualChangeLog = "ManualChangeLog.csv";
         public const string RegistrationLog = "RegistrationLog.csv";
-        public const string EmailSendLog = "EmailSendLog.csv";
```
[the remark this answers is not in the document]
Comment (aidayounesi) [reply GenderPayGap.Core/Enums.cs:296]: All looks good to me, thanks for removing the extra files. Are you happy to merge it?

**Pull request merged**